src/v/cloud_topics/reconciler/reconciler.h

```diff
@@ -42,9 +44,8 @@ class reconciler {
 public:
     reconciler(
       ss::sharded<cluster::partition_manager>*,
-      ss::sharded<cloud_io::remote>*,
       data_plane_api*,
-      std::optional<cloud_storage_clients::bucket_name> = std::nullopt);
+      ss::sharded<l1::file_io>*);
```

Comment:
for duality with `data_plane_api` consider just taking a raw pointer to the shard local member of this `sharded` class